### PR TITLE
feat: added phone number formats

### LINKED ISSUE #84: Phone Number Formats
### Description

Support formats in phone number function.

References:

https://gist.github.com/anubhavshrimal/75f6183458db8c453306f93521e93d37
https://en.wikipedia.org/wiki/National_conventions_for_writing_telephone_numbers

### Proposed solution

_No response_

### Alternatives considered

_No response_

### Do you want to create a pull request?

No

### PR DESCRIPTION
**Pull request by arulprabakaran**
Different phone number formats added based on country code

✅ Closes #84 

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/falso/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Only US phone numbers are generated.

Issue Number: #84 

## What is the new behavior?

Option to generate phone number based on country code is added

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/falso/src/lib/user.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ export function randUser<Options extends FakeOptions>(options?: Options) {
       email: randEmail(),
       firstName: randFirstName(),
       lastName: randLastName(),
-      phone: randPhoneNumber(),
+      phone: <string>randPhoneNumber(),
```
Comment (NetanelBasal) [inline packages/falso/src/lib/user.ts:49]: It should be string, why do you need the casting?

File: packages/falso/src/lib/phone-number.ts [hunk 2]
```diff
@@ -12,15 +246,36 @@ import { randNumber } from './number';
...
+      })?.formats || [];
+  }
+
+  const minNumb = 0;
+  const maxNumb = 9;
 
-    return number.replace(/1(.{3})(.{3})(.{4})/, '1-$1-$2-$3');
-  }, options);
+  let phoneNumber = Array.from({ length: options?.length || 1 }, (_, index) => {
+    return rand(formats).replace(/#/g, () => {
```
Comment (NetanelBasal) [inline packages/falso/src/lib/phone-number.ts:274]: We can use your addition to the `mask` function here, no?

File: packages/falso/src/lib/phone-number.ts [hunk 3]
```diff
@@ -12,15 +246,36 @@ import { randNumber } from './number';
...
+      return (
+        '' + (Math.floor(Math.random() * (maxNumb - minNumb + 1)) + minNumb)
+      );
+    });
+  }) as any;
```
Comment (NetanelBasal) [inline packages/falso/src/lib/phone-number.ts:279]: why as any?

Comment (arulprabakaran) [reply packages/falso/src/lib/phone-number.ts]: My mistake, I must have sensed it when using it in randUser 😁

File: packages/falso/src/lib/phone-number.ts (hunk 2 above)
Comment (arulprabakaran) [reply packages/falso/src/lib/phone-number.ts]: Will do that one `mask` functional changes are merged

File: packages/falso/src/lib/phone-number.ts [hunk 4]
```diff
@@ -12,15 +246,36 @@ import { randNumber } from './number';
...
+        return country.countryCode.includes(options.countryCode!);
+      })?.formats || [];
+  }
+
+  const minNumb = 0;
+  const maxNumb = 9;
 
-    return number.replace(/1(.{3})(.{3})(.{4})/, '1-$1-$2-$3');
-  }, options);
+  let phoneNumber = Array.from({ length: options?.length || 1 }, (_, index) => {
```
[the remark this answers is not in the document]
Comment (arulprabakaran) [reply packages/falso/src/lib/phone-number.ts]: 👍

File: packages/falso/src/lib/phone-number.ts [hunk 5]
```diff
@@ -12,15 +246,36 @@ import { randNumber } from './number';
...
-  options?: Options
-) {
-  return fake(() => {
-    const number = '1' + randNumber({ min: 1_000_000_000, max: 9_999_999_999 });
+export function randPhoneNumber<
+  Options extends FakeOptions & {
+    countryCode?: CountryCode;
+  }
+>(options?: Options) {
+  let formats = data.map(({ formats }) => formats).flat();
```
[the remark this answers is not in the document]
Comment (arulprabakaran) [reply packages/falso/src/lib/phone-number.ts]: 👍

File: packages/falso/src/lib/user.ts (hunk 1 above)
Comment (arulprabakaran) [reply packages/falso/src/lib/user.ts]: It is fixed

File: packages/falso/src/lib/phone-number.ts [hunk 6]
```diff
@@ -12,15 +246,37 @@ import { randNumber } from './number';
...
+  } else {
+    formats = data.map(({ formats }) => formats).flat();
+  }
 
-    return number.replace(/1(.{3})(.{3})(.{4})/, '1-$1-$2-$3');
-  }, options);
+  const phoneNumber = Array.from(
+    { length: options?.length || 1 },
+    (_, index) => {
+      return rand(formats).replace(
```
Comment (NetanelBasal) [inline packages/falso/src/lib/phone-number.ts:275]: Use mask please

Comment (arulprabakaran) [reply packages/falso/src/lib/phone-number.ts]: Done 👍

**Comment by NetanelBasal** [comment]
@arulprabakaran can you please check it?

https://github.com/ngneat/falso/runs/4965354629?check_suite_focus=true

**Pull request merged**